### PR TITLE
TypeScript definitions for i18next

### PR DESCRIPTION
**Pull request by sebastien-comeau**
> i18next has embedded type definitions. If you want to enhance IDE Experience and prevent errors (such as type coercion), you should follow the instructions below in order to get the t function fully-type safe (keys and return type).

https://www.i18next.com/overview/typescript

### CODE REVIEW

File: frontend/app/types/cdcp.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { type RouteHandle as RemixRouteHandle } from '@remix-run/react/dist/routeModules';
+
+import { type Namespace } from 'i18next';
+
+export type RouteHandle = RemixRouteHandle & { i18nNamespaces: Namespace };
```
Comment (gregory-j-baker) [inline frontend/app/types/cdcp.d.ts:5]: I don't know if this makes sense. The data in a route handle depends on the needs of the route.

Not all routes will need an i18n namespace (ie: API routes), and some routes may need to emit different optional data (ie: breadcrumb data).

File: frontend/app/types/cdcp.d.ts
Comment (gregory-j-baker) [inline frontend/app/types/cdcp.d.ts:1]: If we are going to have global application-scoped types, why not just create an `app/types.d.ts` file? Do we need to have a subfolder for this?

File: frontend/@types/i18next.d.ts
Comment (gregory-j-baker) [inline frontend/@types/i18next.d.ts:1]: Why not move this code into `remix.env.d.ts`? I just tested this and it works the same.

image

File: frontend/app/types/cdcp.d.ts
Comment (sebastien-comeau) [reply frontend/app/types/cdcp.d.ts:1]: will fix

File: frontend/@types/i18next.d.ts
Comment (gregory-j-baker) [reply frontend/@types/i18next.d.ts:1]: ex:

``` typescript
import type common from './public/locales/en/common.json';
import type gcweb from './public/locales/en/gcweb.json';

/**
 * @see https://www.i18next.com/overview/typescript
 */
declare module 'i18next' {
  interface CustomTypeOptions {
    resources: {
      common: typeof common;
      gcweb: typeof gcweb;
    };
  }
}

/// <reference types="@remix-run/dev" />
/// <reference types="@remix-run/node" />
```

File: frontend/app/types/cdcp.d.ts
Comment (sebastien-comeau) [reply frontend/app/types/cdcp.d.ts:1]: fixed!

File: frontend/@types/i18next.d.ts
Comment (sebastien-comeau) [reply frontend/@types/i18next.d.ts:1]: > Note that the types referenced in remix.env.d.ts will depend on which environment you're running your app in. For example, there are different globals available in Cloudflare.

https://remix.run/docs/en/main/guides/typescript

Comment (gregory-j-baker) [reply frontend/@types/i18next.d.ts:1]: Your comment doesn't make sense. If we decide to deploy to a non-node environment, then we'll change the type reference.

My point is that we have a file for defining types that aren't included in the standard runtime, so why not use it to define the i18next types?

Comment (sebastien-comeau) [reply frontend/@types/i18next.d.ts:1]: Fixed

File: frontend/app/types/cdcp.d.ts (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/types/cdcp.d.ts]: Fixed! :)

File: frontend/app/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { type Namespace } from 'i18next';
+
+export interface RouteHandle extends Record<string, any> {
```
Comment (gregory-j-baker) [inline frontend/app/types.d.ts:3]:
```suggestion
export interface RouteHandle extends Record<string, unknown> {
```

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

File: frontend/app/types.d.ts (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/app/types.d.ts]: Fixed

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**